### ISSUE TITLE
Implement hooks

### ISSUE THREAD

**Issue by dzeleski** [sensu/sensu-puppet#836]
Currently it appears this module does not support hooks. As this was released in 1.1 I assume others will start wanting to use it at some point. We are trying to start testing them now which is how we ran into it. Basically just wanted to get this open to track the feature, I will attempt to get a PR going if I have some free time later this week.

**Comment by ghoneycutt** [comment]
Thanks for letting us know, much appreciated!

**Comment by alvagante** [comment]
@dzeleski Please give a look to the linked commit and possibly let me know if it does all you need.
I've tested it locally, with sample hooks data similar to the documentation example, and it behaves as expected, but a double check from real world use cases would be welcomed.

**Comment by dzeleski** [comment]
Getting an unrelated error:

`change from 'NT AUTHORITY\SYSTEM' to 'sensu' failed: Could not find group sensu`

`/etc/puppetlabs/code/environments/lab/modules/sensu/manifests/transport.pp:21`

We are currently using an older version of the module, I assume there is a var somewhere that is not getting passed correctly on windows as that group wont work. I have not dug into it just yet but was your branch off of the current masters head?

**Comment by dzeleski** [comment]
Looks like the transport class is not using the platform specific var?

https://github.com/sensu/sensu-puppet/blob/master/manifests/transport.pp#L23

Needs to look something like this:
```
  file { "${sensu::conf_dir}/transport.json":
    ensure  => $ensure,
    owner   => $::sensu::user,
    group   => $::sensu::group,
    mode    => '0440',
    content => inline_template('<%= JSON.pretty_generate(@transport_type_hash) %>'),
    notify  => $::sensu::check_notify,
}
```

**Comment by ghoneycutt** [comment]
@dzeleski please open a separate issue for this. Please use the current version, v2.38.1, before submitting.

**Comment by dzeleski** [comment]
Done, added PR. Have not tested and I will not be able to until first thing tomorrow as I have to leave the office here in a few min.

**Comment by csabo** [comment]
@alvagante dzeleski tested the hook functionality last night and it worked.

**Comment by dzeleski** [comment]
@csabo there are further updates in #837

**Comment by ghoneycutt** [comment]
Thank you everyone for your help getting this implemented!!

Released in v2.39.0
